ludwig/backend/ray.py

```diff
@@ -340,19 +341,26 @@ def run(
         self,
         train_loop_per_worker: Callable,
         config: Dict[str, Any],
-        dataset: Dict[str, Any],
-        data_loader_kwargs: Dict[str, Any],
-        stream_window_size: Dict[str, Union[None, float]],
-        callbacks: List[Any] = [],
+        dataset: Dict[str, Any] = None,
+        data_loader_kwargs: Dict[str, Any] = None,
+        stream_window_size: Dict[str, Union[None, float]] = None,
+        callbacks: List[Any] = None,
     ) -> Tuple[Dict, TorchCheckpoint]:
+        dataset_config = None
+        if dataset is not None:
+            data_loader_kwargs = data_loader_kwargs or {}
+            stream_window_size = stream_window_size or {}
+            dataset_config = self._get_dataset_configs(dataset, stream_window_size, data_loader_kwargs)
+
+        callbacks = callbacks or []
         trainer_cls = HorovodTrainerRay210 if not _ray220 else HorovodTrainer
         trainer = trainer_cls(
             train_loop_per_worker=train_loop_per_worker,
             train_loop_config=config,
             horovod_config=self.backend_config,
             datasets=dataset,
             scaling_config=self.scaling_config,
-            dataset_config=self._get_dataset_configs(dataset, stream_window_size, data_loader_kwargs),
+            dataset_config=dataset_config,
```

Comment:
In what settings might the `dataset_config` be `None`?

Reply:
This one, we don't use the dataset for batch size tuning.

Reply:
Ah I see– just took a closer look at the tune batch size logic. Thanks!